reference/spender.yaml

```diff
@@ -5383,6 +5383,303 @@ components:
+            Specific to per diem related expenses. Start date from which per
+            diem claim is made.
+        ended_at:
+          allOf:
+            - $ref: '#/components/schemas/timestamptz_pst'
+          description: >
+            Specific to per diem related expenses. End date until which per diem
+            claim is made.
+        travel_classes:
+          $ref: '#/components/schemas/travel_classes'
+        locations:
+          $ref: '#/components/schemas/locations'
+        custom_fields:
+          $ref: '#/components/schemas/custom_fields'
+        num_files:
+          type: integer
+          description: |
+            Number of files/reciepts attached to expense
+          example: 4
+        is_matching_ccc:
+          type: boolean
+          description: |
+            Set this field to true if expense matches any card expense
+          default: false
+          example: false
+        mileage_rate_id:
+          allOf:
+            - $ref: '#/components/schemas/fk_integer'
+          description: >
+            Specific to mileage expense. This represents the Mileage's rate id
+            given by Fyle
+
+            during creation of mileage rate for which this expense is created.
+          nullable: true
+          example: 9080
+        per_diem_rate_id:
+          allOf:
+            - $ref: '#/components/schemas/fk_integer'
+          description: >
+            Specific to per diem expense. This represents the per diem's rate id
+            given by Fyle
+
+            during creation of per diem rate for which this expense is created.
+          nullable: true
+          example: 583
+        per_diem_num_days:
+          type: integer
+          nullable: true
+          description: >
+            Specific to per diem related expenses. Number of days for which per
+            diem claim is made.
+          example: 2
+        mileage_calculated_distance:
+          type: number
+          nullable: true
+          description: >-
+            Specific to mileage expense. Distance computed by Fyle for mileage
+            expenses.
+        mileage_calculated_amount:
+          type: number
+          nullable: true
+          description: >-
+            Specific to mileage expense. Amount computed by Fyle for mileage
+            expenses.
+    expense_check_policies_individual_desired_states:
+      type: object
+      properties:
+        add_approver_user_ids:
+          nullable: true
+          type: array
+          items:
+            type: string
+          description: >
+            List of approver ids that the policy calculation suggests adding to
+            report that contains this expense.
+          example:
+            - usBdsdfscl
+            - usLSsdfwqt
+        run_status:
+          nullable: false
+          type: string
+          enum:
+            - VIOLATED_ACTION_SUCCESS
+            - VIOLATED_ACTION_FAILURE
+            - UNVIOLATED
+          description: >
```

Comment:
This represents the status of policy computation for the input expense object against the policy {represented} by `expense_policy_rule_id`. If the state is UNVIOLATED, it means that the expense is not violating this policy. If the state is VIOLATION_ACTION_SUCCESS, it means that the expense is violating this policy and action stated in the object will be applied to the expense when it is saved. However, if the state is VIOLATION_ACTION_FAILURE, it means that this policy is violated for the expense but since some approver was missing or not set up properly, it will fail to take action on the expense when it is saved.